mcs/class/corlib/System.Security.Cryptography.X509Certificates/X509Helper.cs

```diff
@@ -49,6 +49,41 @@ internal static void InstallNativeHelper (INativeCertificateHelper helper)
 				Interlocked.CompareExchange (ref nativeHelper, helper, null);
 		}
 
+		public static X509CertificateImpl InitFromHandle (IntPtr handle)
+		{
+#if MONO_FEATURE_APPLETLS && ONLY_APPLETLS // ONLY_APPLETLS should not support any other option
+			return InitFromHandleApple (handle);
+#elif MONOTOUCH_WATCH
+			throw new PlatformNotSupportedException ();
+#else
+
+#if MONO_FEATURE_APPLETLS // If we support AppleTls, which is the default, and not overriding to legacy
+			if (System.Environment.IsMacOS && Environment.GetEnvironmentVariable ("MONO_TLS_PROVIDER") != "legacy")
+				return InitFromHandleApple (handle);
+#endif
+#if !MOBILE
+			return InitFromHandleCore (handle);
+#elif !MONOTOUCH && !XAMMAC
+			throw new NotSupportedException ();
+#endif
+#endif
```

Comment:
My suggestion:

    #if MONOTOUCH_WATCH
        throw new PlatformNotSupportedException ();
    #else
      #if MONO_FEATURE_APPLETLS
        var use_appletls = true;
        #if !ONLY_APPLETLS
            use_appletls = System.Environment.IsMacOS && Environment.GetEnvironmentVariable ("MONO_TLS_PROVIDER") != "legacy";
        #endif
        if (apple_tls)
            return InitFromHandleApple (handle);
      #endif // MONO_FEATURE_APPLETLS
      #if MOBILE
        throw new NotSupportedException ();
      #else
        return InitFromHandleCore (handle);
      #endif // MOBILE
    #endif // MONOTOUCH_WATCH

The original has a code-path that didn't return a value, in this version all code paths returns a value (or exception).